docs/calendar.md

```diff
@@ -36,6 +36,46 @@ const calendar = new Intl.DateTimeFormat().resolvedOptions().calendar;
 date.withCalendar(calendar).add({ months: 1 });
 ```
 
+### Invariants Across Calendars
+
+The following "invariants" (statements that are always true) hold for all built-in calendars, and SHOULD hold for properly-authored custom calendars:
+
+- `year`, `month`, `day` are always integers that increase as time goes forward
+- `month` and `day` are always positive integers
+- `date.month === 1` during the first month of any year
+- `date.month === date.monthsInYear` during the last month of any year
+- `month` is always continuous (no gaps)
+- Any date can be serialized to an object using only four properties: `{ year, month, day, calendar }`
+
+### Writing Cross-Calendar Code
```

Comment:
I really like this section. I think it's fine here for now. It makes the page a bit long so it could be moved to its own page in the future, but I wouldn't delay the PR on that.